File: Find the Length of the Longest Common Prefix/Find the Length of the Longest Common Prefix.py
# Solution for Find the Length of the Longest Common Prefix in PY

class Solution:
    def longestCommonPrefix(self, arr1, arr2):

        prefixes = set()
        longest = 0

        # Store all prefixes from arr1
        for num in arr1:
            while num > 0 and num not in prefixes:
                prefixes.add(num)
                num //= 10

        # Check arr2 prefixes
        for num in arr2:

            while num > 0 and num not in prefixes:
                num //= 10

            if num > 0:
                longest = max(longest, len(str(num)))

        return longest

File: Find the Length of the Longest Common Prefix/Find the Length of the Longest Common Prefix.java
// Solution for Find the Length of the Longest Common Prefix in JAVA

class Solution {
    public int longestCommonPrefix(int[] arr1, int[] arr2) {

        Set<Integer> prefixes = new HashSet<>();
        int longest = 0;

        // Store all prefixes from arr1
        for (int num : arr1) {
            while (num > 0 && !prefixes.contains(num)) {
                prefixes.add(num);
                num /= 10;
            }
        }

        // Check prefixes in arr2
        for (int num : arr2) {

            while (num > 0 && !prefixes.contains(num)) {
                num /= 10;
            }

            if (num > 0) {
                longest = Math.max(
                    longest,
                    String.valueOf(num).length()
                );
            }
        }

        return longest;
    }
}
